Content.Server/Tabletop/TabletopSystem.cs

```diff
@@ -0,0 +1,188 @@
+
+            draggableComponent.DraggingPlayer = msg.DraggingPlayer;
+
+            if (!draggedEntity.TryGetComponent<SpriteComponent>(out var spriteComponent)) return;
+
+            if (draggableComponent.DraggingPlayer != null)
+            {
+                spriteComponent.Scale = new Vector2(1.25f, 1.25f);
+                spriteComponent.DrawDepth = (int) DrawDepth.Items + 1;
+            }
+            else
+            {
+                spriteComponent.Scale = Vector2.One;
+                spriteComponent.DrawDepth = (int) DrawDepth.Items;
+            }
+        }
+
+        // Remove all tabletop sessions and maps when the round restarts
+        private void OnRoundRestartCleanup(RoundRestartCleanupEvent args)
+        {
+            Reset();
+        }
+
+        #endregion
+
+        #region Utility
+
+        /**
+         * <summary>Create a camera entity for a user to control, and add the user to the view subscribers.</summary>
+         * <param name="user">The user entity to create this camera for and add to the view subscribers.</param>
+         * <param name="coordinates">The map coordinates to spawn this camera at.</param>
+         */
+        // TODO: this can probably be generalized into a "CctvSystem" or whatever
+        private IEntity CreateCamera(IEntity user, MapCoordinates coordinates)
+        {
+            // Spawn an empty entity at the coordinates
+            var camera = _entityManager.SpawnEntity(null, coordinates);
+
+            // Add an eye component and disable FOV
+            var eyeComponent = camera.EnsureComponent<EyeComponent>();
+            eyeComponent.DrawFov = false;
+
+            // Add the user to the view subscribers. If there is no player session, just skip this step
+            if (user.PlayerSession() is { } playerSession)
+            {
+                Get<ViewSubscriberSystem>().AddViewSubscriber(camera.Uid, playerSession);
+            }
+
+            return camera;
+        }
+
+        /**
+         * <summary>Get the <see cref="MapId"/> related to this entity UID.</summary>
+         * <param name="uid">The identifier of the entity to get the map for.</param>
+         * <returns>The <see cref="MapId"/> that has been reserved for this entity.</returns>
+         */
+        private MapId GetMapId(EntityUid uid)
+        {
+            if (_gameSessions.ContainsKey(uid))
+            {
+                return _gameSessions[uid];
+            }
+
+            throw new KeyNotFoundException("The table for the requested entity has not been initialized yet.");
+        }
+
+        /**
+         * <summary>Remove all tabletop sessions and their maps.</summary>
+         */
+        private void Reset()
+        {
+            foreach (var (_, mapId) in _gameSessions)
+            {
+                _mapManager.DeleteMap(mapId);
+            }
+
+            _gameSessions.Clear();
+        }
```

Comment:
You should delete the map whenever the component is shutdown as otherwise these could leak indefinitely. It also means you don't need to listen to RoundRestartEvent as the shutdowns should be raised.